tests/networkpolicy_test.go

```diff
@@ -111,14 +65,12 @@ var _ = Describe("[rfe_id:150][crit:high][vendor:cnv-qe@redhat.com][level:compon
 
 		It("[test_id:1511] should be failed to reach vmia from vmib", func() {
 			By("Connect vmia from vmib")
-			ip := vmia.Status.Interfaces[0].IP
-			assertPingFail(ip, vmib)
+			pingEventually(vmib, vmia.Status.Interfaces[0].IP).ShouldNot(Succeed())
```

Comment:
I think this will not provide the expected result.
Consider a case where the ping fist succeeds (because the policy is not yet deployed) and only after 5sec fails.
In such a case, the eventually will immediately pass and we will fail the assert (when we should have waited instead).

Reply:
the default parameters are -c 1 -w 5 as you know, 
each success will return but as its wrapped in eventually it will retry each 1 seconds (as its the interval of Eventually),
so there are around 15 retries (because the total retry is 15 seconds), 
i saw by practice that 15 seconds is enough and there are no falkes.

Also talked with the community and there isnt a way to validate ingress easily, 
the CNI watch for pods and creates the rules in an async matter

Reply:
Now I see that the helper does not assert, just returns the result and the assertion here will cause a retry. So the scenario I used above will work fine.

Still I am worry here with the introduction of this retry.
Lets check the "Should(Succeed())" scenario: What if it succeeds at the beginning and then blocks when the policy is deployed? We will never detect this in case we actually expect it to pass in the end.
It can also work the other way around: We expect it to fail in general, but it fails only at the begining and when the policy is applied it passes, this is another case where we will miss the problem.

Are there such a scenarios here?

I guess that compared to the current asserts, this retry is not worse, the question is if it is good enough.

Reply:
lets split it to the expect ping fail and expect ping success, for simplicity.

expect pass:
( the scenario you said),
we can increase the "-c 12 -w 3", this way it would check ping for longer time

expect fail:
a policy should stop the traffic,
we are pinging until the first stop, we can add a sanity that 3 stops should occur, i dont think it must be at this very moment imho, i feel what we did is good enough for now, we dont check sanity of checks on at least few other places as well.